intake_xarray/utils.py

```diff
@@ -1,20 +1,57 @@
-def rev_format(format_string, resolved_text):
+def reverse_format(format_string, resolved_string):
     from string import Formatter
+    from datetime import datetime
 
     fmt = Formatter()
     args = {}
-    _text = resolved_text
+    _text = resolved_string
 
-    for i in fmt.parse(format_string):
-        literal_text, field_name, format_spec, conversion = i
-        _text = _text.replace(literal_text, '', 1)
+    # split the string into bits
+    literal_texts, field_names, format_specs, conversions = zip(*fmt.parse(format_string))
+    if not any(field_names):
+        return {}
 
-        if field_name is not None:
-            if not format_spec:
-                raise ValueError('Format spec missing for {}.'.format(field_name))
+    for i, conversion in enumerate(conversions):
+        if conversion:
+            raise ValueError('Conversion not allowed. Found on {}.'.format(field_names[i]))
+
+    # get a list of the parts that matter
```

Comment:
This should probably be its own method.

Reply:
I made this change in https://github.com/ContinuumIO/intake/pull/157

Reply:
@jsignell , does that mean that this PR should now wait until the intake one is merged?